tianshou/data/batch.py

```diff
@@ -36,8 +37,11 @@ def _is_scalar(value: Any) -> bool:
     # 3. python object rather than dict / Batch / tensor
     # the check of dict / Batch is omitted because this only checks a value.
     # a dict / Batch will eventually check their values
-    value = np.asanyarray(value)
-    return value.size == 1 and not value.shape
+    if isinstance(value, (np.ndarray, torch.Tensor)):
+        return value.size == 1 and not value.shape
+    else:
+        value = np.asanyarray(value)
+        return _is_scalar(value)
```

Comment:
Maybe just adding two lines is ok. Why make a recursive call?

Reply:
Maybe something like:
```python
if not isinstance(value, torch.Tensor):
    value = np.asanyarray(value)
return value.size == 1 and not value.shape
```
?

Reply:
It should be OK now.

NB: To be accurate, it is not "recursive" but nested to itself.

Reply:
> It should be OK now.

Forgot to push?